File: go/btree/same-binary-tree.go
package btree

import "fmt"

func isSameTree(p *Node, q *Node) bool {
	if p == nil && q == nil {
		return true
	}
	if p != nil && q != nil && p.Val == q.Val {
		return true && isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
	}
	return false
}

func RunIsSameTree() {
	tre1 := Tree{
		Root: &Node{
			Val: 1,
			Left: &Node{
				Val: 2,
			},
			Right: &Node{
				Val: 3,
				Left: &Node{
					Val: 4,
				},
			},
		},
	}
	tre2 := Tree{
		Root: &Node{
			Val: 1,
			Left: &Node{
				Val: 2,
			},
			Right: &Node{
				Val: 3,
				Left: &Node{
					Val: 4,
				},
			},
		},
	}

	fmt.Println(isSameTree(tre1.Root, tre2.Root))
}

File: javascript/same-binary-tree.js
'use strict';

const sameBinaryTree = (p, q) => {
    if (!p && !q) {
        return true;
    }
    
    if (!p || !q) {
        return false;
    }
    
    if (p.val !== q.val) {
        return false;
    }
    
    return sameBinaryTree(p.left, q.left) && sameBinaryTree(p.right, q.right);
};

const p = {
    val: 1,
    left: {
        val: 2,
        left: null,
        right: null
    },
    right: {
        val: 3,
        left: null,
        right: null
    },
};

const q = {
    val: 1,
    left: {
        val: 2,
        left: null,
        right: null
    },
    right: {
        val: 3,
        left: null,
        right: null
    }
};

console.log(sameBinaryTree(p, q)); // true